### DISCUSSION TITLE
Cycle start after tool change works differently from Ui vs. switch

### DISCUSSION THREAD

**Discussion by samitray** [grblHAL/core#591] [General]
After manual tool change a cycle start from the UI, causes the program to resume. But if  I use the cycle start switch after tool change I get error 40.
Normally i would go for the cycle start switch to avoid greasy stuff on the mouse/track pad. Is this by design or have i missed something
I am on the STM32F4xx driver and custom board.... checked on IO sender manual tool change mode....similar behaviour

**Comment by terjeio** [comment]
Which mode? From the $341 setting.

**Reply by samitray** [reply]
$341=0

**Reply by samitray** [reply]
Earlier I wan not able to get it working in candle, After reading the docks , i figured that ACK needs to be sent, so got IO sender and observed the serial stream, but even with IO sender when I hit the cycle start button, i got error 40. but with cycle start from screen on IO sender, I was able to resume the job....Add the patch to candle, now I can resume from tool change from UI on candle, but not from cycle button.

**Comment by terjeio** [comment]
What is your grblHAL version ($I output).

**Reply by samitray** [reply]
[VER:1.1.20240427:]
[OPT:VNMSL,100,1024,6,16]
[AXS:6:XYZABC]
[NEWOPT:ENUMS, RT+,SED]
[FIRMWARE:grblHAL]
[SIGNALS:RHSP]
[NVS STORAGE:*FLASH]
[FREE MEMORY:102K]
[DRIVER:STM32F446]
[DRIVER VERSION:240418]
[AUX 10:3,3,0,0]
ok

**Reply by terjeio** [reply]
I am not able to replicate this behaviour with the latest version, please update and try again.
